tests/unit/common/eeprom/Eeprom.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "eeprom_acces.h"
+using EepromKey = const char *;
+
+enum class EepromVersion : uint16_t {
+    v1,
+    v2,
+    v3,
+    v4,
+    count = v4
+};
+
+template <class T>
+struct MemConfigBound {
+    T min;
+    T max;
+    T def;
+    bool validate(const T &data) {
+        return (data >= min && data <= max);
+    }
+};
+
+template <class T, class CovertTo = T>
+struct MemConfigItem {
+    EepromAccess &eeprom;
```

Comment:
That's what I'm challenging.

Conceptually, you have independent config items that can update themselves inside the eeprom. You can just hold the item and you're OK. Besides that feeling somewhat strange (an item just living in the ether?), this brings the disadvantage that each single config item will be at least 4 bytes bigger because of the reference. And if I understand it correctly (which I'm not sure of), these are long-living somewhere, or even are the „definitions“ for the values.

What I would think about is eeprom „has“ bunch of items. Or the config store has. So you'd call something like `store.set(...)` and the eeprom would be available in the set.

Reply:
Yes this items are definitions for the items and are also the ram mirror of the eeprom. 

I have changed the reference to the eeprom_access to pointer to the store. 

T think this is good tradeoff between the implementation and the posibility to access the eeprom items as member variables.